lib/stopovers.js

```diff
@@ -0,0 +1,114 @@
+'use strict'
+
+const got = require('got')
+const moment = require('moment-timezone')
+const memoize = require('p-memoize')
+const isString = require('lodash.isstring')
+const isDate = require('lodash.isdate')
+const groupBy = require('lodash.groupby')
+const merge = require('lodash.merge')
+
+const _stations = require('./stations')
+const stations = memoize(_stations, {maxAge: 24*60*60*1000})
+stations()
+
+const defaults = () => ({
+    when: new Date()
+})
+
+const getKeyFromBoth = (obj1, obj2, key) => {
+    let result
+    if (obj1) result = obj1[key]
+    if (obj2) result = obj2[key]
+    if (!result) throw new Error('unexpected result from server, please report this issue')
+    if (obj1 && obj1[key] !== result) throw new Error('unexpected result from server, please report this issue')
+    return result+''
+}
+
+const createStopover = (station, day) => arrivalDepartureTuple => {
+    if (arrivalDepartureTuple.length > 2 || arrivalDepartureTuple.length === 0) throw new Error('unexpected result from server, please report this issue')
+    const arrival = arrivalDepartureTuple.find(t => t.direction === 'ARR')
```

Comment:
Throw if these are missing instead of asserting on `arrivalDepartureTuple` above.